File: src/main/java/com/angorasix/projects/core/presentation/dto/ProjectDto.java
package com.angorasix.projects.core.presentation.dto;

import com.angorasix.projects.core.domain.Attribute;
import com.angorasix.projects.core.domain.project.Requirement;
import java.util.Collection;
import lombok.AllArgsConstructor;
import lombok.Data;

@Data
@AllArgsConstructor
public class ProjectDto {

  private String name;
  private Collection<Attribute<?>> attributes;
  private Collection<Requirement<?>> requirements;

}

File: src/main/kotlin/com/angorasix/projects/core/presentation/dto/ProjectDto.kt
package com.angorasix.projects.core.presentation.dto

import java.time.ZonedDateTime

data class ProjectDto(
    var id: String?,
    var name: String?,
    var attributes: Collection<AttributeDto>?,
    var requirements: Collection<AttributeDto>?,
    var creatorId: String?,
    var createdAt: ZonedDateTime?
) {

    constructor(
        name: String,
        attributes: Collection<AttributeDto> = mutableSetOf<AttributeDto>()
    ) : this(
        null,
        name,
        attributes,
        emptyList(),
        null,
        null
    )
}
